Use CFLAGS and LDFLAGS when compile-chez-program builds itself

weinholt opened the pull request:
This allows a user to pass -lz to the compiler, which is required on Debian.

Admittedly I didn't check if it's harmless to always pass -lz.

gwatt commented:
I had thought about this, and presumably on debian you would need to pass `-lz` every time you run `compile-chez-program`, not just on the initial build. Would it make more sense to have a config option for extra compiler arguments that are always provided?

weinholt commented:
I agree that it would make more sense. I will redo it as a gen-config argument.

Commit:
Additional arguments to gen-config.ss are passed to the C compiler

This adds a way to pass extra arguments to the C compiler that are
used when compile-chez-program links with Chez Scheme. This is useful
on e.g. Debian that needs -lz.

weinholt commented:
Additional arguments to gen-config.ss are now saved to config.ss and appended with compiler-args.
